packages/element-library/src/product/constants.js

```diff
@@ -35,6 +35,8 @@ export const defaultAttributes = {
 
 export const hasEditMode = false;
 
+export const hasDesignMenu = false;
```

Comment:
Just noting that this will change soon-ish in #11252.

But this PR still makes sense to make the design menu more robust, so thanks for tackling this!